charts/mlp/templates/_helpers.tpl

```diff
@@ -0,0 +1,111 @@
+{{/* vim: set filetype=mustache: */}}
+
+
+{{/*
+Generated names
+*/}}
+
+{{- define "mlp.resource-prefix-with-release-name" -}}
+    {{- $deployedChart := .Chart.Name -}}
+    {{- $chartVersion := .Chart.Version | replace "." "-" -}}
+    {{- $deployedReleaseName := .Release.Name -}}
+    {{ printf "%s-%s-%s"  $deployedChart $chartVersion $deployedReleaseName }}
+{{- end -}}
+
+{{- define "mlp.resource-prefix" -}}
+    {{- $deployedChart := .Chart.Name -}}
+    {{- $chartVersion := .Chart.Version | replace "." "-" -}}
+    {{ printf "%s-%s"  $deployedChart $chartVersion }}
+{{- end -}}
+
+
+{{- define "mlp.name" -}}
+    {{- if .Values.nameOverride -}}
+        {{- .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+    {{- else -}}
+        {{- printf "%s" (include "mlp.resource-prefix" .) | trunc 63 | trimSuffix "-" -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "mlp.encryption-key-name" -}}
+    {{- printf "%s-encryption-key" (include "mlp.resource-prefix" .) | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+
+{{- define "mlp.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version -}}
+{{- end -}}
+
+{{- define "mlp.fullname" -}}
+    {{- if .Values.fullnameOverride -}}
+        {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+    {{- else -}}
+        {{- printf "%s" (include "mlp.resource-prefix-with-release-name" .) | trunc 63 | trimSuffix "-" -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "mlp.labels" -}}
+app.kubernetes.io/name: {{ template "mlp.name" . }}
+helm.sh/chart: {{ printf "%s-%s" .Chart.Name .Chart.Version | quote}}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/part-of: caraml
+{{- end }}
+
+{{/*
+Common annotations
+*/}}
+{{- define "mlp.annotations" -}}
+meta.helm.sh/release-name: {{ .Release.Name }}
```

Comment:
shouldn't Helm add these annotations itself?

Reply:
removed, thanks for pointing.